### ISSUE TITLE
Installation instructions

### ISSUE THREAD

**Issue by halx** [MolecularAI/QSARtuna#30]
I think the installation instructions in the README should be presented earlier and tagged more clearly.

It is unclear to me why the Python version has been tied down to the exact patch level.  That doesn't make sense.

Also, the instructions refer to a very specific QSARtuna version.  I would expect this to be valid for at least the major version of the software.

I had to install optuna manually for some reason.  Probably an issue on my side.

**Comment by marco-chimfarm** [comment]
Fully agree with you on the Python version and about QSARtuna dependencies in general.
I'm having serious issues integrating the latest version of Reinvent4 (4.4.22) with the latest version of QSARtuna (3.1.3), even going back to the version 3.0.0.1 does not help.

**Comment by halx** [comment]
@marco-chimfarm You can try the following.  Create a conda environment with python 3.10, install QSARtuna and optuna with pip,  install REINVENT into the same environment and then reinstall scikit-learn in the version QSARtuna requires (at the moment 1.5.1).  I have not tested this much, so other packages, not in the right version, could cause problems: chemprop, protobuf, pandas (pip tells you which versions are "incompatible" with QSARtuna).

**Comment by marco-chimfarm** [comment]
I have tried that as well, but when I install Reinvent with the requirements file it will uninstall and overwrite QSARtuna dependencies and everything is broken again.
What is the command you are using to install Reinvent? Could you share your environment please?

I also just need the Qptuna scoring component, I don't need the full QSARtuna functionalities (at least when associated with reinvent)

**Comment by halx** [comment]
I have installed as below which works with my setup, also on an AMD based machine.  You would need to be more specific as to what is broken and how that manifests itself.

```shell
conda create -n test python=3.10
conda activate test

cd QSARtuna
git pull
pip install .  # after fixing the python version in pyproject.toml
pip install optuna

cd REINVENT4
git pull
pip install -r requirements-linux-64.lock  # using torch 2.4.1
pip install --no-deps .

pip install scikit-learn==1.5.1
```

**Comment by marco-chimfarm** [comment]
Hi @halx, sorry for the delay.
Thanks for sharing your code!
I was quite sure that I had followed your exact order and it didn't work. I've tried your code and now it works perfectly!
Now I'm trying to make it work with Maize as well, let's see if I will have issues...

**Comment by halx** [comment]
As far as REINVENT goes, maize is called though an external process and as such does not need to be installed into the same environment.

**Comment by lewismervin1** [comment]
Hi both, in the latest branch [here ](https://github.com/MolecularAI/QSARtuna/tree/3.1.4_prep) I changed the Poetry versioning for Python to `python = "^3.10"` which I hope will at least relax the Python versioning.

**Comment by halx** [comment]
Many thanks, Lewis.  For general benefit we have also started to relax and review dependencies for REINVENT in the hope that this will ease co-installation with other packages.  The real problem is of course the random mix of packages that conda/pip/etc. are and there is just so much one can do.

**Comment by shortydutchie** [comment]
I ran into the python issue as well when I inadvertently used an older Python version (3.9 I think). One I saw is that the bleeding-edge use of the Annotated typing is a bit restrictive and requires later Python versions. It might be possible if you included

`from __future__ include annotations`

wherever that typing is used. Haven't checked it in anger but that should make things work with slightly earlier Python versions, at least where typing is concerned.

See https://docs.python.org/3/library/__future__.html - the annotation requirement has not been pinned down yet (PEP563).
